lib/charms/operator_libs_linux/v0/snap.py

```diff
@@ -260,24 +267,51 @@ def unset(self, key) -> str:
         """
         return self._snap("unset", [key])
 
-    def _install(self, channel: Optional[str] = "") -> None:
+    def _install(self, channel: Optional[str] = "", cohort: Optional[str] = "") -> None:
         """Add a snap to the system.
 
         Args:
+          confinemenet: strict or classic
           channel: the channel to install from
+          cohort: optional, the key of a cohort that this snap belongs to
         """
         confinement = "--classic" if self._confinement == "classic" else ""
         channel = '--channel="{}"'.format(channel) if channel else ""
-        self._snap("install", [confinement, channel])
+        args = [confinement, channel]
+
+        if not cohort:
+            cohort = self._cohort
+        if cohort:
+            args.append('--cohort="{}"'.format(cohort))
 
-    def _refresh(self, channel: Optional[str] = "") -> None:
+        self._snap("install", args)
+
+    def _refresh(
+        self,
+        channel: Optional[str] = "",
+        cohort: Optional[str] = "",
+        leave_cohort: Optional[bool] = False,
```

Comment:
Could these be combined such that setting `cohort=None` is enough to imply leaving the cohort?

Reply:
Got it - thanks!